content/en/docs/userguide/circuit_breaker.md

```diff
@@ -0,0 +1,253 @@
+---
+draft: false
+linktitle:  Circuit Breaker
+menu:
+  docs:
+    parent: user guide
+    weight: 22
+title: Use Circuit Breaker 
+toc: true
+type: docs
+---
+
+### Preparation
+
+1. Ensure KMesh is installed in your Kubernetes cluster (see [Quick Start Guide](https://kmesh.net/en/docs/setup/quickstart/))
+
+2. Deploy a sample microservice application
+
+3. Verify the default namespace is managed by KMesh
+
+### Circuit Breaker Configuration
+
+##### Deploy a Sample Application
+
+Let's use a simple microservice setup to demonstrate circuit breaking:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sample-service
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: sample-service
+  template:
+    metadata:
+      labels:
+        app: sample-service
+    spec:
+      containers:
+      - name: service
+        image: your-service-image
+        ports:
+        - containerPort: 8080
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: sample-service
+spec:
+  selector:
+    app: sample-service
+  ports:
+  - port: 80
+    targetPort: 8080
+EOF
+```
+
+##### Apply Circuit Breaker Configuration
+
+Configure circuit breaker to limit connections and requests:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: kmesh.net/v1alpha1
+kind: CircuitBreaker
+metadata:
+  name: sample-service-circuit-breaker
+spec:
+  service: sample-service
+  rules:
+    - priority: HIGH
+      maxConnections: 100
+      maxPendingRequests: 50
+      maxRequests: 200
+      maxRetries: 3
+EOF
+```
+
+### Verify Circuit Breaker Functionality
+
+Use a load testing tool to simulate traffic and observe circuit breaker behavior:
+
+```bash
+# Install hey load testing tool
+go install github.com/rakyll/hey@latest
+
+# Generate load
+hey -n 1000 -c 50 http://sample-service/endpoint
```

Comment:
I am not sure how ciercuit breaking work as this is run from a binary, how is the client managed by kmesh